### PR TITLE
add shard & index benchmarks

### PR DESCRIPTION
**Pull request by dgnorton**

### CODE REVIEW

File: tsdb/shard_test.go [hunk 1]
```diff
@@ -73,3 +74,124 @@ func TestShardWriteAndIndex(t *testing.T) {
...
+func BenchmarkWritePoints_ExistingSeries_1M(b *testing.B) {
+	benchmarkWritePointsExistingSeries(b, 320, 5, 5, 1)
+}
+
+func benchmarkWritePoints(b *testing.B, mCnt, tkCnt, tvCnt, pntCnt int) {
+	// Generate test series (measurements + unique tag sets).
+	series := genTestSeries(mCnt, tkCnt, tvCnt)
+	// Load the in-memory series index.
+	index := NewDatabaseIndex()
+	for _, s := range series {
```
Comment (pauldix) [inline tsdb/shard_test.go]: actually, this test should be run without building the in-memory index first. I want to see the total time. The actual saving of the metadata to Bolt won't happen if you do this so it skews the result.

**Comment by dgnorton** [comment]
@pauldix  removed the pre-build of the index.

**Comment by pauldix** [comment]
+1

**Comment by toddboom** [comment]
+1

**Pull request merged**